packages/compass-e2e-tests/tests/search-indexes.test.ts

```diff
@@ -0,0 +1,200 @@
+      await mongoClient.connect();
+      dbInstance = mongoClient.db(DB_NAME);
+
+      // Try to delete a namespace if it exists
+      try {
+        await dbInstance.dropCollection(COLL_NAME);
+      } catch (e) {
+        // noop
+      }
+
+      // Create a namespace
+      await dbInstance.createCollection(COLL_NAME);
+    }
+
+    await browser.connectWithConnectionString(connectionString);
+    await browser.navigateToCollectionTab(DB_NAME, COLL_NAME, 'Indexes');
+  }
+
+  async function afterEachTestRun(currentTest?: CurrentTest) {
+    // Drop the collection
+    {
+      try {
+        await dbInstance.dropCollection(COLL_NAME);
+      } catch (e) {
+        console.log(`Failed to drop collection: ${DB_NAME}.${COLL_NAME}`);
+      }
+    }
+    void mongoClient.close();
+    await disconnect(browser);
+    await afterTest(compass, currentTest);
+  }
+
+  for (const { name, connectionString } of connectionsWithNoSearchSupport) {
+    context(`does not support search indexes in ${name}`, function () {
+      before(function () {
+        if (!connectionString) {
+          return this.skip();
+        }
+      });
+      beforeEach(async function () {
+        await beforeEachTestRun(connectionString!);
+      });
+      afterEach(async function () {
+        await afterEachTestRun(this.currentTest);
+      });
+      it('allows users to create a regular index', async function () {
+        const indexName = await browser.createIndex({
+          fieldName: 'e2e_tests_index',
+          indexType: 'text',
+        });
+        await browser.dropIndex(indexName);
+      });
+      it('renders search indexes tab disabled', async function () {
+        const searchTab = await browser.$(
+          Selectors.indexesSegmentedTab('search-indexes')
+        );
+        const isTabClickable = await searchTab.isClickable();
+        expect(isTabClickable).to.be.false;
+      });
+    });
+  }
+
+  for (const { name, connectionString } of connectionsWithSearchSupport) {
+    context(`supports search indexes in ${name}`, function () {
+      const INDEX_NAME = 'e2e_search_index';
+      const INDEX_DEFINITION = JSON.stringify({
+        mappings: {
+          dynamic: true,
+        },
+      });
+
+      before(function () {
+        if (!connectionString) {
+          return this.skip();
+        }
+      });
+      beforeEach(async function () {
+        await beforeEachTestRun(connectionString!);
+
+        // Try to drop the existing search indexes. As we started afresh,
+        // the collection should not have any search index. However this
+        // is to just avoid any race condition.
+        {
+          const searchIndexes = await dbInstance
+            .collection(COLL_NAME)
+            .listSearchIndexes()
+            .toArray();
+
+          if (searchIndexes.length > 0) {
+            await Promise.all(
+              searchIndexes.map(({ name }) =>
+                dbInstance.collection(COLL_NAME).dropSearchIndex(name)
```

Comment:
Won't this run other CI tasks' search indexes? Maybe we should rather create random/unique search index names and then in an afterEach clean them up. We then also don't have to wait for them to be deleted because it will happen after the tests end and we won't care about the name being available because it is random and unique.